src/Servers/Kestrel/Core/src/Internal/Http/HttpProtocol.cs

```diff
@@ -659,7 +659,7 @@ public void OnTrailersComplete()
             var messageBody = CreateMessageBody();
             if (!messageBody.RequestKeepAlive)
             {
-                _keepAlive = false;
+                DisableKeepAlive(ConnectionEndReason.RequestNoKeepAlive);
```

Comment:
@halter73 I've updated HTTP/1.1 connection layer to set an end reason when keep alive is set to false. I think the end reasons could be improved. Especially `ResponseNoKeepAlive` which I just added in a bunch of places when writing the server is writing the response.